### PR TITLE
Add .match() for just matching and returning routes

### PR DESCRIPTION
**Pull request by shama**
Here are the basic changes needed to have `base-router` run using `wayfarer`.

This allows me to do:

```js
var router = wayfarer()

// Check if the route exists without calling the callback
var matched = router.match(name)
if (!matched) done(new Error('404'))

// Call the callback with custom params and inspect the return
try {
  var data = matched.node.cb(params, done)
  // is data a promise or just return value?
} catch (err) {
  done(err)
}
```

Also it changes the `const` to `var` which makes it compatible with IE and node.js.

I figured this was an easier way to show the changes needed. Thanks for the consideration!

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
+1, seems like a reasonable interface to consume `wayfarer` in other routers / modules. Thanks!

File: test.js [hunk 1]
```diff
@@ -52,3 +52,14 @@ test('aliases', function (t) {
   const r = wayfarer()
   t.equal(r, r.emit)
 })
+
+test('.match()', function (t) {
+  t.plan(2)
+  const r = wayfarer()
+  r.on('/user/:id', function () {
+    t.fail('route should not be called with .match()')
```
Comment (yoshuawuyts) [inline test.js]: `coveralls` doesn't appear to be a fan of `t.fail` - https://coveralls.io/builds/2740508/source?filename=test.js :/

**Comment by yoshuawuyts** [comment]
Thanks! Published as `wayfarer@3.1.0` :tada:

File: test.js (hunk 1 above)
Comment (shama) [inline test.js]: Oh dang sorry, I'll rework the test and fix.

**Comment by shama** [comment]
Super rad! Thanks a bunch!

**Pull request merged**